connector_clickup/models/clickup_backend/common.py

```diff
@@ -0,0 +1,364 @@
+                if from_date:
+                    from_date = from_date.strftime("%Y-%m-%dT%H:%M:%SZ")
+                else:
+                    from_date = to_iso_datetime(from_date)
+
+            filters["with_count"] = "true"
+            force = False
+            if force_update_field:
+                force = backend[force_update_field]
+
+            model_parts = model.split(".")
+            model_name = " ".join(part.title() for part in model_parts[1:])
+            model_name = " ".join(dict.fromkeys(model_name.split()))
+            job_options["description"] = f"Export Batch of Clickup {model_name}"
+
+            clickup_model = (
+                self.env[model].with_delay(**job_options or {})
+                if with_delay
+                else self.env[model]
+            )
+
+            clickup_model.export_batch(
+                backend,
+                filters=filters,
+            )
+            if force:
+                backend[force_update_field] = False
+        next_time = import_start_time - timedelta(seconds=IMPORT_DELTA_BUFFER)
+        if from_date_field:
+            self.write({from_date_field: next_time})
+
+    def cron_import_clickup_changes(
+        self,
+        filters=None,
+        with_delay=True,
+    ):
+        """Scheduled action method of import"""
+        data = self.env["clickup.backend"].search([])
+
+        for backend in data:
+            clickup_model = backend.env["clickup.project.project"].with_delay()
+            clickup_model.import_batch(
+                backend,
+                filters=filters,
+                force=True,
+                **{"no_delay": not with_delay},
+            )
+        for backend in data:
+            clickup_model = backend.env["clickup.project.task.type"].with_delay()
+            clickup_model.import_batch(
+                backend,
+                filters=filters,
+                force=True,
+                **{"no_delay": not with_delay},
+            )
+        for backend in data:
+            clickup_model = backend.env["clickup.project.tasks"].with_delay()
+            clickup_model.import_batch(
+                backend,
+                filters=filters,
+                force=True,
+                **{"no_delay": not with_delay},
+            )
+
+    def cron_export_clickup_changes(
+        self,
+        filters=None,
+        with_delay=True,
+    ):
+        """Scheduled action method of export"""
+        data = self.env["clickup.backend"].search([])
+
+        for backend in data:
+            clickup_model = backend.env["clickup.project.project"].with_delay()
+            clickup_model.export_batch(
+                backend,
+                filters=filters,
+            )
+        for backend in data:
+            clickup_model = backend.env["clickup.project.tasks"].with_delay()
+            clickup_model.export_batch(
+                backend,
+                filters=filters,
+            )
+
+    def generate_token(self):
```

Comment:
I think we should introduce the cron to generate_token in every half hour, or before 10 min. of expiration time.

Reply:
I get the same access token every time I generate it. 
https://clickup.com/api/developer-portal/faq/#do-oauth-access-tokens-expire